docs/tutorials/infer_phys_pars.rst

```diff
@@ -0,0 +1,503 @@
+relations, offset by :math:`180^\circ` in :math:`\Omega_\mathrm{p}`. We use
+Astropy's :py:class:`~astropy.coordinates.Angle` and
+:py:meth:`~astropy.coordinates.Angle.wrap_at` to restrict the values of
+:math:`\Omega_\mathrm{p}` to its allowed range of
+:math:`0^\circ \leq \Omega_\mathrm{p} < 360^\circ`.
+
+.. jupyter-execute::
+
+    i_p = np.linspace(0., 180., 181) << u.deg
+
+    delta_omega_p1 = np.arctan(np.tan(chi_p) / np.cos(i_p))
+    delta_omega_p2 = delta_omega_p1 + 180.*u.deg
+
+    omega_p1 = xi - delta_omega_p1
+    omega_p2 = xi - delta_omega_p2
+
+    omega_p1 = Angle(omega_p1).wrap_at(360.*u.deg).deg * u.deg
+    omega_p2 = Angle(omega_p2).wrap_at(360.*u.deg).deg * u.deg
+
+The two :math:`i_\mathrm{p}`--:math:`\Omega_\mathrm{p}` relations are
+disjointed at :math:`i_\mathrm{p} = 90^\circ`, where
+:math:`\cos( i_\mathrm{p} )` changes sign. For plotting, we stitch the four
+halves of the two solutions together appropriately to create two continuous
+curves in :math:`i_\mathrm{p}`--:math:`\Omega_\mathrm{p}` space.
+
+.. jupyter-execute::
+
+    ii_ccw = (i_p <= 90.*u.deg)
+    ii_cw =  (i_p >  90.*u.deg)
+
+    omega_p_stitch1 = np.concatenate((omega_p1[ii_ccw], omega_p2[ii_cw]))
+    omega_p_stitch2 = np.concatenate((omega_p2[ii_ccw], omega_p1[ii_cw]))
+
+.. jupyter-execute::
+
+    plt.figure(figsize=(7., 6.))
+
+    plt.plot(i_p, omega_p_stitch1, c='C0')
+    plt.plot(i_p, omega_p_stitch2, c='C0')
+
+    plt.xlim(0., 180.)
+    plt.ylim(0., 360.)
+
+    plt.xlabel(r"pulsar's orbital inclination $i_\mathrm{p}$")
+    plt.ylabel(r"pulsar's longitude of ascending node $\Omega_\mathrm{p}$")
+
+    plt.show()
+
+Plotting the relations shows how :math:`\Omega_\mathrm{p}` is restricted to two
+ranges of values (while :math:`i_\mathrm{p}` is still unrestricted).
+
+.. jupyter-execute::
+
+    print(f'{omega_p_stitch1[-1].to(u.deg):.2f} < omega_p < '
+          f'{omega_p_stitch1[0].to(u.deg):.2f}    or    '
+          f'{omega_p_stitch2[-1].to(u.deg):.2f} < omega_p < '
+          f'{omega_p_stitch2[0].to(u.deg):.2f}')
+
+
+The effective distance
+----------------------
+
+Next, the effective distance :math:`d_\mathrm{eff}` can be calculated using
+
+.. math::
+
+    d_\mathrm{eff} = \frac{ v_\mathrm{orb,E}^2 }{ A_\mathrm{E}^2 }
+                     b_\mathrm{E}^2.
+
+
+.. jupyter-execute::
+
+    b2_e = (1 - np.sin(i_e)**2) / (1 - np.sin(i_e)**2 * np.cos(chi_e)**2)
+    d_eff = v_orb_e**2 / amp_e**2 * b2_e
+
+    print(f'd_eff:   {d_eff.to(u.pc):8.2f}')
+
+
+Given the effective distance, it is possible to derive a relation between
+the distance to the pulsar :math:`d_\mathrm{p}` and the distance to the screen
+:math:`d_\mathrm{s}`. In terms of the fractional screen--pulsar distance
+:math:`s`, the two true distances are given by
+
+.. math::
+
+    d_\mathrm{p} &= \frac{ s }{ 1 - s } d_\mathrm{eff}, \\
+    d_\mathrm{s} &= s d_\mathrm{eff}.
+
+.. jupyter-execute::
+
+    ns = 250
+    s = np.arange(0.5/ns, 1., 1./ns)
```

Comment:
I'd prefer `np.arange(0, ns)+0.5) / ns` - `arange` on float can have unpleasant rounding errors.

Reply:
Switched to ``linspace`` instead.